python/GafferSceneUI/EditScopeUI.py

```diff
@@ -154,3 +157,150 @@ def __visibilityKeyPress( viewer, event ) :
 				tweakPlug["value"].setValue( False )
 
 	return True
+
+
+# Processor Widgets
+# =================
+
+class _SceneProcessorWidget( GafferUI.EditScopeUI.SimpleProcessorWidget ) :
+
+	def _linkActivated( self, linkData ) :
+
+		GafferSceneUI.ContextAlgo.setSelectedPaths(
+			self.processor().ancestor( Gaffer.ScriptNode ).context(), linkData
+		)
+
+class __LocationEditsWidget( _SceneProcessorWidget ) :
+
+	@staticmethod
+	def _summary( processor, linkCreator ) :
+
+		# Get the locations being edited from the spreadsheet.
+
+		canceller = Gaffer.Context.current().canceller()
+		activePathMatcher = IECore.PathMatcher()
+		disabledPathMatcher = IECore.PathMatcher()
+		for row in processor["edits"] :
+			IECore.Canceller.check( canceller )
+			path = row["name"].getValue()
+			if not path :
+				continue
+			if row["enabled"].getValue() :
+				activePathMatcher.addPath( path )
+			else :
+				disabledPathMatcher.addPath( path )
+
+		# Match those against the scene.
+
+		activePaths = IECore.PathMatcher()
+		disabledPaths = IECore.PathMatcher()
+		GafferScene.SceneAlgo.matchingPaths( activePathMatcher, processor["in"], activePaths )
+		GafferScene.SceneAlgo.matchingPaths( disabledPathMatcher, processor["in"], disabledPaths )
+
+		# Build a summary describing what we found.
+
+		summaries = []
+		if activePaths.size() :
+			activeLink = linkCreator(
+				"{} location{}".format( activePaths.size(), "s" if activePaths.size() > 1 else "" ),
+				activePaths
+			)
+			summaries.append( f"edits on {activeLink}" )
+		if disabledPaths.size() :
+			disabledLink = linkCreator(
+				"{} location{}".format( disabledPaths.size(), "s" if disabledPaths.size() > 1 else "" ),
+				disabledPaths
+			)
+			summaries.append( f"disabled edits on {disabledLink}" )
+
+		if not summaries :
+			return "None"
+
+		summaries[0] = summaries[0][0].upper() + summaries[0][1:]
+		return " and ".join( summaries )
+
+GafferUI.EditScopeUI.ProcessorWidget.registerProcessorWidget( "AttributeEdits TransformEdits *LightEdits *SurfaceEdits", __LocationEditsWidget )
+
+class __PruningEditsWidget( _SceneProcessorWidget ) :
+
+	@staticmethod
+	def _summary( processor, linkCreator ) :
+
+		paths = IECore.PathMatcher()
+		GafferScene.SceneAlgo.matchingPaths( processor["PathFilter"], processor["in"], paths )
+
+		if paths.isEmpty() :
+			return "None"
+		else :
+			link = linkCreator(
+				"{} location{}".format( paths.size(), "s" if paths.size() > 1 else "" ),
+				paths
+			)
+			return f"{link} pruned"
+
+GafferUI.EditScopeUI.ProcessorWidget.registerProcessorWidget( "PruningEdits", __PruningEditsWidget )
+
+class __RenderPassesWidget( GafferUI.EditScopeUI.SimpleProcessorWidget ) :
+
+	@staticmethod
+	def _summary( processor, linkCreator ) :
+
+		names = processor["names"].getValue()
+		return "{} render pass{} created {}".format(
+			len( names ) if names else "No",
+			"es" if len( names ) > 1 else "",
+			"( {} )".format( ", ".join( names ) ) if names else "",
```

Comment:
With a non-trivial amount of "production" render pass names, this can pretty quickly turn into an unhelpfully long summary that pushes all of the enabled and show widgets off screen. Maybe it's worth dropping this summary text if there isn't a straightforward way of eliding it while keeping the other widgets visible?

![longRenderPassNames](https://github.com/GafferHQ/gaffer/assets/50844517/72f37088-b8cc-417a-b6f3-318a7440fa42)

Reply:
Good point - for now I've just dropped the summary text, since QLabel doesn't provide any easy way of eliding text. Happy to iterate on this later, but I think it's worth getting the current version into a release already. Also added an API entry to Changes.md and rebased to get this on top of current `main`.